src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java

```diff
@@ -0,0 +1,153 @@
+package io.github.thebusybiscuit.slimefun4.core.services;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.plugin.Plugin;
+import org.bukkit.plugin.java.JavaPlugin;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import io.github.thebusybiscuit.slimefun4.core.debug.Debug;
+import io.github.thebusybiscuit.slimefun4.core.debug.TestCase;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+
+/**
+ * This class represents an analytics service that sends data.
+ * This data is used to analyse performance of this {@link Plugin}.
+ * <p>
+ * You can find more info in the README file of this Project on GitHub.
+ *
+ * @author WalshyDev
+ */
+public class AnalyticsService {
+
+    private static final int VERSION = 1;
+    private static final String API_URL = "https://analytics.slimefun.dev/ingest";
+
+    private final JavaPlugin plugin;
+    private final HttpClient client = HttpClient.newHttpClient();
+
+    private boolean enabled;
+
+    public AnalyticsService(JavaPlugin plugin) {
+        this.plugin = plugin;
+    }
+
+    public void start() {
+        this.enabled = Slimefun.getCfg().getBoolean("metrics.analytics");
+
+        if (enabled) {
+            plugin.getLogger().info("Enabled Analytics Service");
+
+            // Send the timings data every minute
+            Slimefun.getThreadService().newScheduledThread(
+                plugin,
+                "AnalyticsService - Timings",
+                sendTimingsAnalytics(),
+                1,
+                1,
+                TimeUnit.MINUTES
+            );
+        }
+    }
+
+    // Timings we'll send the average every minute
```

Comment:
Confused about what this means but may be cause I am tired

Reply:
That should be will iirc

Reply:
I wrote this while tired too 😅

Reply:
```suggestion
    // We'll send some timing data every minute,
    // Today, we collect; the tick interval, the avg timing per tick and avg timing per machine
```